File: Kotlin/src/monteCarloPi.kt
import java.util.Random

/**
 * Created by tuckerstrausbaugh on 11/8/16.
 */

fun main(args: Array<String>) {
    val start = System.currentTimeMillis()
    val pi: Double = (piLoop(args[0].toInt()).toDouble() / args[0].toDouble()) * 4.0
    val end = System.currentTimeMillis()

    println("pi: $pi")
    println("time: " + (end - start) / 1000.0 + "s")
}

fun monteCarloPi(reps: Int): Int {
    var count = 0

    for (i in 1..reps)
        if (inUnitCircle(Random().nextDouble(), Random().nextDouble()))
            count++

    return count
}

fun inUnitCircle(x: Double, y: Double): Boolean {
    return if (x*x + y*y < 1.0) true else false
}

File: Python/monte_carlo_pi.py
import random
from sys import argv


def inUnitCircle (x: float, y: float):
    if x*x + y*y < 1.0:
        return True
    else:
        return False


def monteCarloPi (reps: int):
    count = 0

    for rep in range(reps):
        if inUnitCircle(random.uniform(0.0, 1.0), random.uniform(0.0, 1.0)):
            count += 1

    return count


reps = int(argv[1])
count = monteCarloPi(reps)
pi = count / reps * 4

print("pi: " + str(pi))
